File: csharp/DesiArchitect.DesignPatterns/01_Strategy/After/UpiPaymentStrategy.cs
using DesiArchitect.DesignPatterns.Shared;

namespace DesiArchitect.DesignPatterns.Strategy.After;

public class UpiPaymentStrategy : IPaymentStrategy
{
    public PaymentResult Pay(decimal amount)
    {
        Console.WriteLine($"  [After] UPI payment of ₹{amount} processed via NPCI gateway");
        return new PaymentResult(true, "UPI payment successful");
    }
}

File: typescript/src/01-strategy/after/upi-payment-strategy.ts
import type { IPaymentStrategy } from './payment-strategy.js';
import type { PaymentResult } from '../../shared/models.js';

export class UpiPaymentStrategy implements IPaymentStrategy {
  pay(amount: number): PaymentResult {
    console.log(`  [After] Charging ₹${amount} via UPI strategy`);
    return { success: true, message: 'UPI payment processed' };
  }
}
